modules/@apostrophecms/schema/index.js

```diff
@@ -1048,15 +1049,15 @@ module.exports = {
             object.scopedArrayName = field.scopedArrayName;
           },
           object: (field, object) => {
-            if (!can(field)) {
+            if (!object || !can(field)) {
               return;
             }
 
             object.metaType = 'object';
             object.scopedObjectName = field.scopedObjectName;
           },
           relationship: (field, doc) => {
-            if (!can(field)) {
+            if (!Array.isArray(doc[field.name]) || !can(field)) {
```

Comment:
I handle empty values directly in `prepareForStorage` because in the simulate case I want to be able to enter in the handler when there is no value.